src/ranged.cpp

```diff
@@ -265,27 +250,44 @@ bool player::handle_gun_damage( item &it )
             }
         }
     }
-    if( curammo_effects.count( "BLACKPOWDER" ) ) {
-        if( !it.faults.count( fault_gun_blackpowder ) &&
-            it.faults_potential().count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_blackpowder );
+    if( !curammo_effects.count( "NON-FOULING" ) ) {
+        if( curammo_effects.count( "BLACKPOWDER" ) ) {
+            if( ( it.ammo_data()->ammo->recoil < firing->min_cycle_recoil ) &&
```

Comment:
It was intentional that this failed to cycle on every shot when the ammunition was blackpowder.
We need some more evidence for which guns can cycle with blackpowder.

Reply:
I think it's just an optical illusion on GitHub's change analyzer, Kevin. All I did was add:

if( !curammo_effects.count( "NON-FOULING" ) ) {

to that big section dealing with blackpowder stuff, but because it applies to the entire section, the entire section was indented. So GitHub thinks I deleted that whole section and then added it back, but in reality all I did was insert that one line more or less and the astyle indented it.

The purpose of that line, btw, is to allow people to add a NON-FOULING parameter to non-gunpowder ammo (like the rails for the rail rifle) to avoid fouling of guns like the rail rifle, which makes no sense since it uses electromagnetism/electricity and NOT gunpowder as propellant.

Speaking of which, I think I need to do the same for the nail guns/air-based weapons to prevent fouling in these non-GP guns.

Reply:
BP ammo should still cause a malfunction, btw, unless something broke.

Reply:
I just checked and BP still causes stovepiping, so I didn't remove that functionality.

Reply:
Yea my bad, sorry.